protobuf/flow/access/access.proto

```diff
@@ -424,6 +451,55 @@ message GetAccountAtBlockHeightRequest {
   uint64 block_height = 2;
 }
 
+message GetAccountBalanceAtLatestBlockRequest {
+  bytes address = 1;
+}
+
+message GetAccountBalanceAtBlockHeightRequest {
+  bytes address = 1;
+  uint64 block_height = 2;
+}
+
+message AccountBalanceResponse {
+  uint64 balance = 1;
+  uint64 available_balance = 2;
+  entities.Metadata metadata = 3;
+}
+
+message GetAccountKeysAtLatestBlockRequest {
+  // address of account
+  bytes address = 1;
+}
+
+message GetAccountKeyAtLatestBlockRequest {
+  // address of account
+  bytes address = 1;
+  // index of key to return
+  entities.AccountKeyIndex index = 2;
```

Comment:
since we're using separate endpoints for `*Key` vs `*Keys`, you can flatten this to just a `uint32`. It will simplify the logic for clients.